crm_salesperson_planner/models/crm_salesperson_planner_visit_template.py

```diff
@@ -0,0 +1,397 @@
+        string="Recurrence Termination",
+        default="count",
+    )
+    interval = fields.Integer(
+        string="Repeat Every", default=1, help="Repeat every (Days/Week/Month/Year)"
+    )
+    count = fields.Integer(string="Repeat", help="Repeat x times", default=1)
+    mo = fields.Boolean(string="Mon")
+    tu = fields.Boolean(string="Tue")
+    we = fields.Boolean(string="Wed")
+    th = fields.Boolean(string="Thu")
+    fr = fields.Boolean(string="Fri")
+    sa = fields.Boolean(string="Sat")
+    su = fields.Boolean(string="Sun")
+    month_by = fields.Selection(
+        [("date", "Date of month"), ("day", "Day of month")],
+        string="Option",
+        default="date",
+    )
+    day = fields.Integer(string="Date of month", default=1)
+    week_list = fields.Selection(
+        [
+            ("MO", "Monday"),
+            ("TU", "Tuesday"),
+            ("WE", "Wednesday"),
+            ("TH", "Thursday"),
+            ("FR", "Friday"),
+            ("SA", "Saturday"),
+            ("SU", "Sunday"),
+        ],
+        string="Weekday",
+    )
+    byday = fields.Selection(
+        [
+            ("1", "First"),
+            ("2", "Second"),
+            ("3", "Third"),
+            ("4", "Fourth"),
+            ("5", "Fifth"),
+            ("-1", "Last"),
+        ],
+        string="By day",
+    )
+    last_visit_date = fields.Date(
+        string="Last Visit Date", compute="_compute_last_visit_date", store=True
+    )
+    final_date = fields.Date(string="Repeat Until")
+
+    _sql_constraints = [
+        (
+            "crm_salesperson_planner_visit_template_name",
+            "UNIQUE (name)",
+            "The name must be unique!",
+        ),
+    ]
+
+    @api.model
+    def create(self, values):
+        if values.get("name", "/") == "/":
+            values["name"] = self.env["ir.sequence"].next_by_code(
+                "salesperson.planner.visit.template"
+            )
+        res = super().create(values)
+        return res
+
+    @api.depends("visit_ids.date")
+    def _compute_last_visit_date(self):
+        for sel in self.filtered(lambda x: x.visit_ids):
+            sel.last_visit_date = sel.visit_ids.sorted(lambda x: x.date)[-1].date
+        self.filtered(lambda x: not x.visit_ids).write({"last_visit_date": False})
+
+    def _compute_visit_ids_count(self):
+        for sel in self:
+            sel.visit_ids_count = len(sel.visit_ids)
+
+    def action_view_salesperson_planner_visit(self):
+        action = self.env.ref(
+            "crm_salesperson_planner.all_crm_salesperson_planner_visit_action"
+        ).read()[0]
+        action["domain"] = [("id", "=", self.visit_ids.ids)]
+        action["context"] = {
+            "default_partner_id": self.partner_id.id,
+            "default_visit_template_id": self.id,
+            "default_description": self.description,
+        }
+        return action
+
+    # --- Recurrency ---
+
+    def _rrule_serialize(self):
```

Comment:
All of this is a verbatim copy of the `calendar.event` code that is included in Odoo core, and you must reuse it instead of duplicating code that may have unnoticed bugs since the moment you perform the fork. One possible thing is to do inheritance of the model changing the final name.

Reply:
I thought about doing an inheritance of the model to reuse it, but there are really many things that are not needed and the original functions are not required all, so I was cleaning them and changing them a bit to adapt it to what I need.

Reply:
No matter if there are things not used. You are not copying code and you only use what you want.

Reply:
Sorry, I had not seen that it was already being discussed here.
I do not know why these comments do not appear in the conversation tab...

Reply:
@carlosdauden : this model don't use datetime, only date and priority. The calendar.event use datetime for show the records in the calendar views.
@pedrobaeza : okay.

both: I will look at the way you tell me to use it. Thanks for the comments.

Reply:
I see convenient to order visits inside same day, but if not, you can do it through a calendar.event marking the event as "all day event".

Reply:
The problem would come when he had more than one visit per day (which is normal). It would fill his entire calendar and it would be difficult to manage if he has to schedule other actions.

Reply:
But that visits are time that you spent, and it should be reflected. Anyways, if they are put as all day, they are folded on the up part.

Reply:
We have put it together with calendar.event, what do you think?